docs/source/dataframe-hive.rst

```diff
@@ -0,0 +1,227 @@
+    ['part.0.parquet', 'part.1.parquet']
+
+
+It is important to recognize that Dask will **not** aggregate the
+data files written within each of the leaf directories. This is
+because each of the DataFrame partitions is written independently 
+during the execution of the :func:`to_parquet` task graph. In order 
+to write out data for partition `i`, the partition-i write task will
+perform a `groupby` operation on columns ``["year", "semester"]``,
+and then each distinct group will be written to the corresponding
+directory using the file name ``'part.{i}.parquet'``. Therefore, it
+is possible for a hive-partitioned write to produce a large number
+of files in every leaf directory (one for each DataFrame partition).
+
+
+Reading Parquet Data with Hive Partitioning
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In most cases, :func:`read_parquet` will process hive-partitioned
+data automatically. By default, all hive-partitioned columns will
+be interpreted as categorical columns.
+
+.. code-block:: python
+
+    >>> ddf = dd.read_parquet("output-path", columns=["year", "semester"])
+
+    >>> ddf
+    Dask DataFrame Structure:
+                            year         semester
+    npartitions=4                                  
+                category[known]  category[known]
+                            ...              ...
+                            ...              ...
+                            ...              ...
+                            ...              ...
+    Dask Name: read-parquet, 1 graph layer
+
+    >>> ddf.compute()
+    year semester
+    0  2022     fall
+    1  2022     fall
+    2  2022     fall
+    3  2022   spring
+    4  2022   spring
+    5  2022   spring
+    6  2023     fall
+    7  2023     fall
+
+
+Defining a Custom Partitioning Schema
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When utilizing ``engine='pyarrow'``, it is possible to specify a custom
+schema for the hive-partitioned columns.
+
+.. code-block:: python
+
+    >>> schema = pa.schema([("year", pa.int16()), ("semester", pa.string())])
+
+    >>> ddf2 = dd.read_parquet(
+    ...     path,
+    ...     columns=["year", "semester"],
+    ...     dataset={"partitioning": {"flavor": "hive", "schema": schema}}
+    ... )
+    Dask DataFrame Structure:
+                    year semester
+    npartitions=4                
+                int16   object
+                    ...      ...
+                    ...      ...
+                    ...      ...
+                    ...      ...
+
+
+If any of your hive-partitioned columns contain null values, you
+**must** specify the partitioning schema in this way.
+
+Although it is not required, we also recommend that you specify
+the partitioning schema if you need to partition on high-cardinality
+columns. This is because the default ``'category'`` dtype will
+track the known categories in a way that can significantly increase
+the overall memory footprint of your Dask collection.
```

Comment:
Slight off-topic. I assume the answer is no but do we have docs on this subject we could link to? I believe known vs unknown categories is a topic that would be worth a dedicated page/section somewhere since this is all but obvious for a novice user.

Reply:
I guess this is sort of mentioned in [the Categoricals doc page](https://docs.dask.org/en/stable/dataframe-categoricals.html). We say over there that a `to_parquet`/`read_parquet` round-trip will "forget" known categories to avoid a similar performance problem. It probably makes sense to add that link here.